staging/src/k8s.io/client-go/openapi/groupversion.go

```diff
@@ -38,9 +39,23 @@ func newGroupVersion(client *client, item handler3.OpenAPIV3DiscoveryGroupVersio
 }
 
 func (g *groupversion) Schema(contentType string) ([]byte, error) {
-	return g.client.restClient.Get().
-		RequestURI(g.item.ServerRelativeURL).
-		SetHeader("Accept", contentType).
-		Do(context.TODO()).
-		Raw()
+	locator, err := url.Parse(g.item.ServerRelativeURL)
+	if err != nil {
+		return nil, err
+	}
+
+	path := g.client.restClient.Get().
+		AbsPath(locator.Path).
+		SetHeader("Accept", contentType)
+
+	// Other than root endpoints(openapiv3/apis), resources have hash query parameter to support etags.
+	// However, absPath does not support handling query parameters internally,
+	// so that hash query parameter is added manually
+	for k, value := range locator.Query() {
```

Comment:
Another way would be defining a new function `AbsPathWithQuery` in request package and use;

```go
return g.client.restClient.Get().
		AbsPathWithQuery(g.item.ServerRelativeURL).
		SetHeader("Accept", contentType).
		Do(context.TODO()).
		Raw()
```